includes/admin/home/class-sensei-home-remote-data-provider.php

```diff
@@ -0,0 +1,133 @@
+<?php
+/**
+ * File containing Sensei_Home_Remote_Data_Provider class.
+ *
+ * @package sensei-lms
+ * @since   $$next-version$$
+ */
+
+/**
+ * Class responsible for fetching data from SenseiLMS.com.
+ */
+class Sensei_Home_Remote_Data_Provider {
+	const API_BASE_URL     = 'https://senseilms.com/wp-json/senseilms-home/1.0/';
+	const CACHE_KEY_PREFIX = 'sensei_home_remote_data_';
+	const CACHE_TTL        = DAY_IN_SECONDS;
+
+	/**
+	 * The primary plugin to pull data for.
+	 *
+	 * @var string
+	 */
+	private $primary_plugin_slug;
+
+	/**
+	 * Sensei_Home_Remote_Data_Provider constructor.
+	 *
+	 * @param string $primary_plugin_slug The primary plugin slug.
+	 */
+	public function __construct( string $primary_plugin_slug ) {
+		$this->primary_plugin_slug = $primary_plugin_slug;
+	}
+
+	/**
+	 * Fetch data from SenseiLMS.com.
+	 *
+	 * @param int $max_age Maximum age of the cached data in seconds. Max is 1 day (in seconds).
+	 *
+	 * @return array|\WP_Error
+	 */
+	public function fetch( int $max_age = null ) {
+		$url           = $this->get_api_url();
+		$transient_key = self::CACHE_KEY_PREFIX . md5( $url );
+		$data          = get_transient( $transient_key );
+
+		// If the data is too old, fetch it again.
+		if ( $max_age && is_array( $data ) ) {
```

Comment:
I am struggling a little to understand why we need an age check as well as the cache TTL. Can't we simplify this and just rely on the cache TTL value?
In case we want more up-to-date information reducing the `$max_age` will do the same as reducing the cache TTL.

Maybe I am missing some part 😅

Reply:
I think it's because it's one day to fetch for the counter in the WP Admin menu (coming through a CRON IIRC), and it's 1 hour when we access directly the home. But maybe Jake can answer better. Sorry for intruding! haha

Reply:
Actually, thinking more, if that's the reason and we will use CRON to fetch it daily, maybe we could use 1 hour for every case. Because it will run once a day anyway. Unless I missed something. 🤔

Reply:
> Actually, thinking more, if that's the reason and we will use CRON to fetch it daily, maybe we could use 1 hour for every case. Because it will run once a day anyway. Unless I missed something. 🤔

Yup, that's what I meant with the second part. Having two different times doesn't really makes sense to me but let's see if we are missing something 😅

Reply:
Hey! Sorry, I should have added a comment. I thought when fetching for the Sensei Home _badge_ (probably updated each WP Admin request, not cron, just to keep it simple), we can cache for 1 day. However, I thought when they actually visit Sensei Home we potentially want more recent data.

Reply:
> Hey! Sorry, I should have added a comment. I thought when fetching for the Sensei Home badge (probably updated each WP Admin request, not cron, just to keep it simple), we can cache for 1 day. However, I thought when they actually visit Sensei Home we potentially want more recent data.

To be fair you mentioned this in some previous conversations... but somehow I still struggled to make sense of it. Not it makes more sense: `$max_age` is 1h when loading Sensei_Home in order to make data fresher than the max TTL (1 day).

Just to be sure, the part where you call `fetch()` without a 1h max age, is not implemented in this PR yet. Right? (I couldn't find it).